docker-app/qfieldcloud/core/tests/test_team.py

```diff
@@ -0,0 +1,501 @@
+        )
+
+    def test_admin_user_can_update_team_member_username(self):
+        """Test that an admin can update a team member's username."""
+
+        team_member = TeamMember.objects.create(team=self.t1, member=self.u3)
+
+        self.assertEqual(team_member.member.username, self.u3.username)
+
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token2.key)
+
+        new_username = "updated_username"
+
+        response = self.client.patch(
+            f"/api/v1/organizations/{self.org1.username}/teams/{self.t1.teamname}/members/{self.u3.username}/",
+            {"member": new_username},
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        team_member.refresh_from_db()
+        self.assertEqual(team_member.member.username, new_username)
+
+    def test_non_admin_user_cannot_update_team_member_username(self):
+        """Test that a non-admin user cannot update a team member's username."""
+
+        TeamMember.objects.create(team=self.t1, member=self.u3)
+
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token3.key)
+
+        new_username = "updated_username"
+        response = self.client.patch(
+            f"/api/v1/organizations/{self.org1.username}/teams/{self.t1.teamname}/members/{self.u3.username}/",
+            {"username": new_username},
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        team_member = TeamMember.objects.get(team=self.t1, member=self.u3)
+        self.assertEqual(team_member.member.username, self.u3.username)
+
+    def test_unauthorized_user_cannot_update_team_member_username(self):
+        """Test that an unauthorized user cannot update a team member's username."""
+
+        TeamMember.objects.create(team=self.t1, member=self.u3)
+
+        new_username = "updated_username"
+        response = self.client.patch(
+            f"/api/v1/organizations/{self.org1.username}/teams/{self.t1.teamname}/members/{self.u3.username}/",
+            {"username": new_username},
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        team_member = TeamMember.objects.get(team=self.t1, member=self.u3)
+        self.assertEqual(team_member.member.username, self.u3.username)
+
+    def test_admin_user_cannot_update_team_member_with_empty_payload(self):
+        team_member = TeamMember.objects.create(team=self.t1, member=self.u3)
+
+        self.assertEqual(team_member.member.username, self.u3.username)
+
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token2.key)
+
+        response = self.client.patch(
+            f"/api/v1/organizations/{self.org1.username}/teams/{self.t1.teamname}/members/{self.u3.username}/",
+            {},
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        team_member.refresh_from_db()
+        self.assertEqual(team_member.member.username, self.u3.username)
+
```

Comment:
```suggestion

```

Does not really make sense to update the username. One can simply delete and add a new user.